File: algorithms/sorting/bubbleSort.py
"""
Algorithm for bubble sort
"""

def bubble_sort(arr):

    n = len(arr)

    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[i], arr[j] = arr[j], arr[i]
            else:
                continue

    return arr


print(bubble_sort([4,2,6,7,8,12,5,19]))

File: algorithms/CPP/Sorting/bubbleSort.cpp
#include<iostream>
using namespace std;
void swap(int *xp, int *yp){
  int temp = *xp;
  *xp = *yp;
  *yp = temp;
}

void bubbleSort(int arr[], int n){
  for(int i =0; i< n-1; i++){
    for(int j = 0; j <n-i-1; j++){
      if (arr[j] > arr[j+1]){
        swap(&arr[j], &arr[j+1]);
      }
    }
  }
}

void printArray(int arr[], int size){
  for(int i = 0; i < size; i++){
    cout << arr[i] << " ";
  }
  cout << " " << endl;
}

int main(){
  int arr[] = {2,1,3,5};
  int n = sizeof(arr)/sizeof(arr[0]);
  bubbleSort(arr,n);
  printf("Sorted Array using bubble sort is ");
  printArray(arr, n);
  return 0;
}
